src/components/v5/common/ActionsContent/partials/SocialLinksTable/partials/SocialLinksModal.tsx

```diff
@@ -0,0 +1,121 @@
+import React, { FC } from 'react';
+import clsx from 'clsx';
+import { formatText } from '~utils/intl';
+
+import Modal from '~v5/shared/Modal';
+import { useActionSidebarContext } from '~context/ActionSidebarContext';
+import Button from '~v5/shared/Button';
+import { useMobile } from '~hooks';
+import { useSocialLinks, useSocialLinksButtons } from '../hooks';
+import Input from '~v5/common/Fields/Input';
+import { LINK_TYPE, SocialLinksModalProps } from '../types';
+
+const displayName =
+  'v5.common.ActionsContent.partials.SocialLinlsTable.partials.SocialLinksModal';
+
+const SocialLinksModal: FC<SocialLinksModalProps> = ({
+  fieldArrayMethods,
+  name,
+}) => {
+  const isMobile = useMobile();
+  const {
+    socialLinksModalToggle: [
+      isSocialLinksModalOpen,
+      { toggleOff: toggleChangeSocialLinksModalOff },
+    ],
+  } = useActionSidebarContext();
+  const { formState, register, onFormSubmit, linkName } = useSocialLinks(
+    fieldArrayMethods,
+    name,
+  );
+  const socialLinksButtons = useSocialLinksButtons(linkName);
+
+  return (
+    <Modal
+      isOpen={isSocialLinksModalOpen}
+      onClose={() => toggleChangeSocialLinksModalOff()}
+      buttonMode="primarySolid"
+      icon="smile"
+    >
+      <form onSubmit={onFormSubmit}>
+        <h5 className="heading-5 mb-1.5">
+          {formatText({ id: 'editColony.socialLinks.modal.title' })}
+        </h5>
+        <p className="text-md text-gray-600 mb-6">
+          {formatText({ id: 'editColony.socialLinks.modal.subtitle' })}
+        </p>
+        <div
+          className={clsx('gap-4', {
+            'flex flex-col': isMobile,
+            'grid grid-cols-3': !isMobile,
+          })}
+        >
+          {socialLinksButtons.map((button) => (
+            <Button
+              isFullSize
+              mode="socialLinkType"
+              className="group relative"
+              isFilled={button.type === linkName}
+              key={button.type}
+            >
+              {button.icon}
+              <input
+                key={button.type}
+                type="radio"
+                {...register('name')}
+                value={button.type}
+                className="min-w-[8.375rem] h-[2.5rem] absolute opacity-0 cursor-pointer"
+              />
+              {formatText({ id: `socialLink.button.${button.type}` })}
+            </Button>
+          ))}
+        </div>
+        {linkName && (
+          <div className="mt-6">
+            {linkName === LINK_TYPE.Twitter ? (
+              <>
+                <span className="text-gray-700 text-1 mb-2 capitalize">
```

Comment:
Some styles are repeated. You can add a wrapper and apply those styles there. Also, why is this margin added here?